## *********************************************************
## 
## File autogenerated for the MST_Position package 
## by the dynamic_reconfigure package.
## Please do not edit.
## 
## ********************************************************/

##**********************************************************
## Software License Agreement (BSD License)
##
##  Copyright (c) 2008, Willow Garage, Inc.
##  All rights reserved.
##
##  Redistribution and use in source and binary forms, with or without
##  modification, are permitted provided that the following conditions
##  are met:
##
##   * Redistributions of source code must retain the above copyright
##     notice, this list of conditions and the following disclaimer.
##   * Redistributions in binary form must reproduce the above
##     copyright notice, this list of conditions and the following
##     disclaimer in the documentation and/or other materials provided
##     with the distribution.
##   * Neither the name of the Willow Garage nor the names of its
##     contributors may be used to endorse or promote products derived
##     from this software without specific prior written permission.
##
##  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
##  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
##  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
##  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
##  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
##  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
##  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
##  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
##  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
##  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
##  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
##  POSSIBILITY OF SUCH DAMAGE.
##**********************************************************/

from dynamic_reconfigure.encoding import extract_params

inf = float('inf')

config_description = {'upper': 'DEFAULT', 'lower': 'groups', 'srcline': 233, 'name': 'Default', 'parent': 0, 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'cstate': 'true', 'parentname': 'Default', 'class': 'DEFAULT', 'field': 'default', 'state': True, 'parentclass': '', 'groups': [], 'parameters': [{'srcline': 259, 'description': 'bool to use the garmin gps', 'max': True, 'cconsttype': 'const bool', 'ctype': 'bool', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'use_gpsd', 'edit_method': '', 'default': False, 'level': 1, 'min': False, 'type': 'bool'}, {'srcline': 259, 'description': 'bool to use the garmin gps', 'max': True, 'cconsttype': 'const bool', 'ctype': 'bool', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'continue_when_done', 'edit_method': '', 'default': False, 'level': 1, 'min': False, 'type': 'bool'}, {'srcline': 259, 'description': 'option to go forward or follow waypoints', 'max': True, 'cconsttype': 'const bool', 'ctype': 'bool', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'go_to_waypoints', 'edit_method': '', 'default': False, 'level': 1, 'min': False, 'type': 'bool'}, {'srcline': 259, 'description': 'radius considerd to reach waypoints', 'max': 10.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'waypoint_radius', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'radius considerd to reach waypoints', 'max': 10.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'dummy_point_radius', 'edit_method': '', 'default': 3.0, 'level': 1, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'reverses order of movement', 'max': True, 'cconsttype': 'const bool', 'ctype': 'bool', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'reverse_order', 'edit_method': '', 'default': False, 'level': 1, 'min': False, 'type': 'bool'}, {'srcline': 259, 'description': 'resets waypoint list to begining', 'max': True, 'cconsttype': 'const bool', 'ctype': 'bool', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'reset_waypoints', 'edit_method': '', 'default': False, 'level': 1, 'min': False, 'type': 'bool'}, {'srcline': 259, 'description': 'option to skip a waypoint', 'max': True, 'cconsttype': 'const bool', 'ctype': 'bool', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'skip_waypoint', 'edit_method': '', 'default': False, 'level': 1, 'min': False, 'type': 'bool'}, {'srcline': 259, 'description': 'stops the robot and pauses waypoints', 'max': True, 'cconsttype': 'const bool', 'ctype': 'bool', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'pause', 'edit_method': '', 'default': False, 'level': 1, 'min': False, 'type': 'bool'}, {'srcline': 259, 'description': 'offset aplied to gps heading', 'max': 3.141592653589793, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'heading_offset', 'edit_method': '', 'default': 0.0, 'level': 1, 'min': -3.141592653589793, 'type': 'double'}, {'srcline': 259, 'description': 'option use dummy gps for testing', 'max': True, 'cconsttype': 'const bool', 'ctype': 'bool', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'use_dummy', 'edit_method': '', 'default': True, 'level': 1, 'min': False, 'type': 'bool'}, {'srcline': 259, 'description': 'latitude for dummy gps', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'dummy_latitude', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': -90.0, 'type': 'double'}, {'srcline': 259, 'description': 'longitude for dummy gps', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'dummy_longitude', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': -90.0, 'type': 'double'}, {'srcline': 259, 'description': 'heading for dummy gps in radians', 'max': 3.141592653589793, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'dummy_heading', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': -3.141592653589793, 'type': 'double'}, {'srcline': 259, 'description': 'latitude for waypoint', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_1_latitude', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': -90.0, 'type': 'double'}, {'srcline': 259, 'description': 'longitude for waypoint', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_1_longitude', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': -90.0, 'type': 'double'}, {'srcline': 259, 'description': 'priority of waypoint(0 is ignored)', 'max': 10, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_1_priority', 'edit_method': '', 'default': 1, 'level': 1, 'min': 0, 'type': 'int'}, {'srcline': 259, 'description': 'timelimit for waypoint(0 is ignored) in min', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_1_limit', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'latitude for waypoint', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_2_latitude', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': -90.0, 'type': 'double'}, {'srcline': 259, 'description': 'longitude for waypoint', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_2_longitude', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': -90.0, 'type': 'double'}, {'srcline': 259, 'description': 'priority of waypoint(0 is ignored)', 'max': 10, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_2_priority', 'edit_method': '', 'default': 1, 'level': 1, 'min': 0, 'type': 'int'}, {'srcline': 259, 'description': 'timelimit for waypoint(0 is ignored) in min', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_2_limit', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'latitude for waypoint', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_3_latitude', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': -90.0, 'type': 'double'}, {'srcline': 259, 'description': 'longitude for waypoint', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_3_longitude', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': -90.0, 'type': 'double'}, {'srcline': 259, 'description': 'priority of waypoint(0 is ignored)', 'max': 10, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_3_priority', 'edit_method': '', 'default': 1, 'level': 1, 'min': 0, 'type': 'int'}, {'srcline': 259, 'description': 'timelimit for waypoint(0 is ignored) in min', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_3_limit', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'latitude for waypoint', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_4_latitude', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': -90.0, 'type': 'double'}, {'srcline': 259, 'description': 'longitude for waypoint', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_4_longitude', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': -90.0, 'type': 'double'}, {'srcline': 259, 'description': 'priority of waypoint(0 is ignored)', 'max': 10, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_4_priority', 'edit_method': '', 'default': 1, 'level': 1, 'min': 0, 'type': 'int'}, {'srcline': 259, 'description': 'timelimit for waypoint(0 is ignored) in min', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_4_limit', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'latitude for waypoint', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_5_latitude', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': -90.0, 'type': 'double'}, {'srcline': 259, 'description': 'longitude for waypoint', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_5_longitude', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': -90.0, 'type': 'double'}, {'srcline': 259, 'description': 'priority of waypoint(0 is ignored)', 'max': 10, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_5_priority', 'edit_method': '', 'default': 1, 'level': 1, 'min': 0, 'type': 'int'}, {'srcline': 259, 'description': 'timelimit for waypoint(0 is ignored) in min', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_5_limit', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'latitude for waypoint', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_6_latitude', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': -90.0, 'type': 'double'}, {'srcline': 259, 'description': 'longitude for waypoint', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_6_longitude', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': -90.0, 'type': 'double'}, {'srcline': 259, 'description': 'priority of waypoint(0 is ignored)', 'max': 10, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_6_priority', 'edit_method': '', 'default': 1, 'level': 1, 'min': 0, 'type': 'int'}, {'srcline': 259, 'description': 'timelimit for waypoint(0 is ignored) in min', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_6_limit', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'latitude for waypoint', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_7_latitude', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': -90.0, 'type': 'double'}, {'srcline': 259, 'description': 'longitude for waypoint', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_7_longitude', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': -90.0, 'type': 'double'}, {'srcline': 259, 'description': 'priority of waypoint(0 is ignored)', 'max': 10, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_7_priority', 'edit_method': '', 'default': 1, 'level': 1, 'min': 0, 'type': 'int'}, {'srcline': 259, 'description': 'timelimit for waypoint(0 is ignored) in min', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_7_limit', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'latitude for waypoint', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_8_latitude', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': -90.0, 'type': 'double'}, {'srcline': 259, 'description': 'longitude for waypoint', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_8_longitude', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': -90.0, 'type': 'double'}, {'srcline': 259, 'description': 'priority of waypoint(0 is ignored)', 'max': 10, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_8_priority', 'edit_method': '', 'default': 1, 'level': 1, 'min': 0, 'type': 'int'}, {'srcline': 259, 'description': 'timelimit for waypoint(0 is ignored) in min', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_8_limit', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'latitude for waypoint', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_9_latitude', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': -90.0, 'type': 'double'}, {'srcline': 259, 'description': 'longitude for waypoint', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_9_longitude', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': -90.0, 'type': 'double'}, {'srcline': 259, 'description': 'priority of waypoint(0 is ignored)', 'max': 10, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_9_priority', 'edit_method': '', 'default': 1, 'level': 1, 'min': 0, 'type': 'int'}, {'srcline': 259, 'description': 'timelimit for waypoint(0 is ignored) in min', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_9_limit', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'latitude for waypoint', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_10_latitude', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': -90.0, 'type': 'double'}, {'srcline': 259, 'description': 'longitude for waypoint', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_10_longitude', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': -90.0, 'type': 'double'}, {'srcline': 259, 'description': 'priority of waypoint(0 is ignored)', 'max': 10, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_10_priority', 'edit_method': '', 'default': 1, 'level': 1, 'min': 0, 'type': 'int'}, {'srcline': 259, 'description': 'timelimit for waypoint(0 is ignored) in min', 'max': 90.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/opt/ros/fuerte/stacks/dynamic_reconfigure/src/dynamic_reconfigure/parameter_generator.py', 'name': 'way_10_limit', 'edit_method': '', 'default': 1.0, 'level': 1, 'min': 0.0, 'type': 'double'}], 'type': '', 'id': 0}

min = {}
max = {}
defaults = {}
level = {}
type = {}
all_level = 0

#def extract_params(config):
#    params = []
#    params.extend(config['parameters'])    
#    for group in config['groups']:
#        params.extend(extract_params(group))
#    return params

for param in extract_params(config_description):
    min[param['name']] = param['min']
    max[param['name']] = param['max']
    defaults[param['name']] = param['default']
    level[param['name']] = param['level']
    type[param['name']] = param['type']
    all_level = all_level | param['level']

